### ISSUE TITLE
Cheat Engine 7.5 Build in Lazarus. Identifier not found ExtractFileNameWithoutExt

### ISSUE THREAD

**Issue by AWSchneider-dev** [cheat-engine/cheat-engine#2699]
I keep getting this error when trying to build version 7.5. 

CPU: Ryzen 5 2600
16gb RAM

Followed instructions from github to build the project.

Full error: cesupport line: 194 ExtractFileNameWithoutExt

**Comment by AWSchneider-dev** [comment]
in version 7.4 I get the error frmAAEditPrefsUnit Line: 50 Identifier not found TSynEditorOptions

**Comment by fade2gray** [comment]
Read the build instructions again and install the correct version of Lazarus.

**Comment by gladichl** [comment]
I used lazarus-2.2.2-fpc-3.2.2-win64.exe and then lazarus-2.2.2-fpc-3.2.2-cross-i386-win32-win64.exe from the link in the project and got the same errors. By the way EXE files CheatEngine are marked as dangerous on sourceforce and trainers are always generated by containing trojans. It's very suspicious that on purpose (?) we can't compile Cheat Engine ourselves.

**Comment by fade2gray** [comment]
https://github.com/cheat-engine/cheat-engine/assets/1332009/de261b7b-2e14-43c5-a5b6-ae11f95477b5

**Comment by gladichl** [comment]
It's impossible to compile with this Lazarus version (on Windows 11 at least). By accident I tried 32 bit version and it's the only way to compile Cheat Engine.

**Comment by fade2gray** [comment]
I use Win11 and my video shows it works. If it's not working for you, then you've either messed up the Lazarus installation or you're just doing something wrong. There are plenty of guides on the net on how to compile CE.

**Comment by gladichl** [comment]
From your video, it's not clear that you did it on Windows 11, on which build, or that Lazarus was indeed 64-bit, because they're quite similar (64 bit and 32 bit). You could have used the old version without realizing it. I haven't messed up anything; it's almost impossible to do so because the installation is quite straightforward, and the person above encountered the same thing. I assume that you read the previous texts inattentively, including the part where I indicated that I succeeded with another version of Lazarus. There was a discussion on the Internet that in some cases (or even maybe in many cases) only the 32-bit version works.

**Comment by cheat-engine** [comment]
you got the wrong version of the sourcecode

**Comment by gladichl** [comment]
No, I downloaded the latest 7.5 (the folder "cheat-engine-7.5"). Did you read the title of this topic?

**Comment by fade2gray** [comment]
@gladichl Just for you https://youtu.be/3LI0nPn_9RA
I suggested previously, I think your Lazarus installation may be messed up - try uninstalling, making sure to delete any configuration files, and then reinstalling.

**Comment by gladichl** [comment]
Again, you didn't show what must have been shown. Which Windows build did you use?

**Comment by fade2gray** [comment]
> Again, you didn't show what must have been shown.

What!?

The method outlined in the video will work for 64-bit Windows 10 and 11.

**Comment by gladichl** [comment]
Read attentively what is written below, if you don't mind, because it seems that you don't understand some sentences.
No, it seems that it doesn't work for every Windows version. Possibly for some Windows versions Lazarus 2.2.2 32-bit works only for Cheat Engine 7.5, but it doesn't work for Cheat Engine 7.4. I tried Lazarus 2.2.2 pure 64-32 bit and 32-bit and they both don't work for either Cheat Engine versions (7.5 and 7.4) (I again reinstalled it instead of Lazarus 32 bit to check twice because of you, and it was a stupid idea because nowadays it's quite improbable that something goes wrong during the installation, any possible errors are self-corrected). I get the same error as the above user ("TSynEditorOptions" for Cheat Engine 7.4 in case of Lazarus 64-32 bit and Lazarus 32 bit and "ExtractFileNameWithoutExt" for Cheat Engine 7.5, if Lazarus 64-32 bit is used, Lazarus 32 bit can compile it). Your Windows version matters to me and other potential users. Now it's your problem, not mine. Live with it, if you don't want to provide information which is very simple to provide. It seems that you assume that everybody around you are children, not adults, who can't unpack very simple programs to install (for clarification: not simple programs but simple to install).

**Comment by gladichl** [comment]
Read attentively what is written below, if you don't mind, because it seems that you don't understand some sentences.
No, it seems that it doesn't work for every Windows version. Possibly for some Windows versions Lazarus 2.2.2 32-BIT works only for Cheat Engine 7.5, but it doesn't work for Cheat Engine 7.4. I tried Lazarus 2.2.2 pure 64-32 BIT and it doesn't work for either Cheat Engine versions (7.5 and 7.4) (I again reinstalled it instead of Lazarus 32 BIT to check twice because of you, and it was a stupid idea because nowadays it's quite improbable that something goes wrong during the installation, any possible errors are self-corrected). I get the same error as the above user ("TSynEditorOptions" for Cheat Engine 7.4 in case of Lazarus 64-32 BIT and Lazarus 32 BIT and "ExtractFileNameWithoutExt" for Cheat Engine 7.5, if Lazarus 64-32 BIT is used, Lazarus 32 BIT can compile it). Your Windows version matters to me and other potential users. Now it's your problem, not mine. Live with it, if you don't want to provide information which is very simple to provide. It seems that you assume that everybody around you are children, not adults, who can't unpack very simple programs to install (for clarification: not simple programs but simple to install).

**Comment by fade2gray** [comment]
Consider yourself muted.

**Comment by gladichl** [comment]
Because of your response, I consider that you again didn't understand almost anything that was written or are pretending not to understand. Don't respond because it doesn't make any sense to talk to you; I am writing for other potential users who might encounter the same problem.

Probably fade2gray doesn't even realize that Windows considers this program a hack tool (of course, it's a joke) and that a licensed Windows can protect everything. My recommendation is not to use Lua (especially); it's quite unsafe (I will explain maybe later). If fade2gray doesn't use advanced anti-virus options, it's his/her problem, but he/she doesn't have to write nonsense that Cheat Engine can be compiled on every Windows 10/11 using Lazarus 64-32 BIT, because it's untrue, and I confirm it with the above user. Otherwise, why was I able to compile it using Lazarus 32-bit and wasn't be able to do so twice by using Lazarus 64 BIT? Possibly fade2gray doesn't have any logical explanation regarding this fact.

He/she provided nonsensical information that I didn't even ask for (proofs are above). As you can see from my point of view he/she behaves quite arrogantly and, most importantly, foolishly toward everybody (me and other users). Right now, because of his/her mostly nonsensical comments, I must check if Cheat Engine works on another version of licensed Windows by installing and reinstalling every program. If I don't succeed in doing so, I will write a detailed post about what is going on with this unsafe tool because I have already found something pretty strange in the behavior of its code.

**Comment by gladichl** [comment]
Because of your response, I consider that you again didn't understand almost anything that was written or are pretending not to understand. Don't respond because it doesn't make any sense to talk to you; I am writing for other potential users who might encounter the same problem.

Probably fade2gray doesn't even realize that a licensed Windows considers this program a hack tool (of course, it's a joke) and that a licensed Windows can protect everything. My recommendation is not to use Lua (especially); it's quite unsafe (I will explain maybe later). If fade2gray doesn't use advanced anti-virus options, it's his/her problem, but he/she doesn't have to write nonsense that Cheat Engine can be compiled on every Windows 10/11 using Lazarus 64-32 BIT, because it's untrue, and I confirm it with the above user. Otherwise, why was I able to compile it using Lazarus 32-bit and wasn't be able to do so twice by using Lazarus 64 BIT? Possibly fade2gray doesn't have any logical explanation regarding this fact.

He/she provided nonsensical information that I didn't even ask for (proofs are above). As you can see from my point of view he/she behaves quite arrogantly and, most importantly, foolishly toward everybody (me and other users). Right now, because of his/her mostly nonsensical comments, I must check if Cheat Engine works on another version of licensed Windows by installing and reinstalling every program. If I don't succeed in doing so, I will write a detailed post about what is going on with this unsafe tool because I have already found something pretty strange in the behavior of its code.

**Comment by Akarius** [comment]
I got the same error while compiling it with the exact right version of Lazarus but it happens that I was using the master code, I switch to 7.5 tag and it worked.
